homeassistant/components/deconz/deconz_event.py

```diff
@@ -0,0 +1,60 @@
+"""Representation of a deCONZ remote."""
+from homeassistant.const import CONF_EVENT, CONF_ID
+from homeassistant.core import EventOrigin, callback
+from homeassistant.helpers.device_registry import CONNECTION_ZIGBEE
+from homeassistant.util import slugify
+
+from .const import _LOGGER, DOMAIN
+
+
+class DeconzEvent:
+    """When you want signals instead of entities.
+
+    Stateless sensors such as remotes are expected to generate an event
+    instead of a sensor entity in hass.
+    """
+
+    def __init__(self, device, gateway):
+        """Register callback that will be used for signals."""
+        self._device = device
+        self.gateway = gateway
+
+        self._device.register_async_callback(self.async_update_callback)
+
+        self.event = f"deconz_{CONF_EVENT}"
```

Comment:
What does this do?

Reply:
We shouldn't use configuration key as part of the event that is fired. Can't we just hardcode it to `deconz_remote_pressed` ?

Reply:
I can't change this. I just broke out the deConz event implementation to its own file. Using deconz event is well established. I can make it more readable though.

Reply:
yeah just do 
```suggestion
        self.event = "deconz_event"
```